### PR TITLE
Standalone Parser

### PR DESCRIPTION
**Pull request by robinmayerhofer**
In Endpoints, there may be separate types for parsing, and representing the output.  
In the current version, and the possible usage, a type may be a `ResponseParser` as well as the `ResponseParser.OutputType`.

This is convenient, but does not make sense semantically. Also, there was no way to use a custom `jsonDecoder` for an array of `Decodables` (implicit conformance to `JSONDecodableParser`).

The changes in this PR aim at separating the `ResponseParser` from it's `OutputType`.

See Migration/V2_0_0.md for a changelog and migration guide.

### CODE REVIEW

**Review by mariohahn — APPROVED** [review]
Find ich jetzt viel besser, ist zwar etwas zu adaptieren aber dafür ist es eine neuer Versionssprung

File: Migration/V2_0_0.md [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+### Call
+
+* `Call.ResponseType` was renamed to `Call.Parser`. This does also affect all subtypes.
+
+* Migrate:
+- use `YourCall.Parser` instead of `YourCall.ResponseType`
+
+### Convenience
+
+#### EmptyResponseParser
```
Comment (tkoller) [inline Migration/V2_0_0.md]: Naming-vorschlag: Vielleicht NoContentParser? 
oder hast an besseren Vorschlag?

**Review by tkoller — CHANGES_REQUESTED** [review]
Ansonsten gefällts mir sehr gut.

File: Migration/V2_0_0.md (hunk 1 above)
Comment (robinmayerhofer) [reply Migration/V2_0_0.md]: Yes, `NoContentParser` is likely the better choice, as one primary use case is for 204 responses, which are called No Content. Will update!

Comment (tkoller) [reply Migration/V2_0_0.md]: nice

Comment (robinmayerhofer) [reply Migration/V2_0_0.md]: Done

**Review by mariohahn — APPROVED** [review]

**Review by tkoller — APPROVED** [review]

**Pull request merged**